.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,22 @@
+
+name: Integration tests
+
+on:
+  pull_request:
+  push:
```

Comment:
C'est utile de les jouer au push avec le nouveau fonctionnement ?
Si j'ai bien compris le deploy se fera sans les attendre de toute manière.
Après, peut être mieux vaut être informé du problème même si le deploy a été fait... Mais normalement quand ca se joue sur un event `pull_request` ça se joue sur le résultat du merge, et non sur la branche courante - d'ou le fait qu'il ne se déclenche pas en cas de conflits j'imagine (car pas possible du coup).

Reply:
alors en fait c'est assez bien fait car le deploy se déclenche uniquement si tous les autres checks sont verts

<img width="1043" alt="Capture d’écran 2021-07-21 à 21 14 13" src="https://user-images.githubusercontent.com/2269165/126546417-48e26a1c-1d2f-4df4-a33e-0b42fca34f03.png">

Reply:
> Mais normalement quand ca se joue sur un event `pull_request` ça se joue sur le résultat du merge, et non sur la branche courante - d'ou le fait qu'il ne se déclenche pas en cas de conflits j'imagine (car pas possible du coup)

C'était ma première intuition mais je n'en étais pas certain, si c'est le cas c'est parfait, on peut effectivement se dispenser de faire tourner les tests lorsqu'on pousse sur `dev` et `master` en changeant les params de protection de ces deux branches pour autoriser un merge uniquement si les tests sont verts

Reply:
<img width="415" alt="Capture d’écran 2021-07-21 à 21 30 51" src="https://user-images.githubusercontent.com/2269165/126548497-d110b807-5c4b-47e0-a9c1-36a042794269.png">

Yes tu as raison ça se joue sur le résultat du merge !

Reply:
Du coup j'ai changé le fonctionnement pour faire en sorte que les tests unitaires (`build-and-test`) et les tests d'intégration tournent uniquement sur les événements `pull_request`. J'aurais aimé que les tests unitaires tournent également lorsque qu'on pousse simplement une branche mais je n'ai pas trouvé comment ne pas faire tourner lestests deux fois quand on pousse sur une PR ouverte dans la configuration suivante : 

```yaml 
on: 
  push:
    branches_ignore:
      - dev 
      -  master
   pull_request:
```